src/Illuminate/Support/Arr.php

```diff
@@ -331,7 +331,19 @@ public static function isAssoc(array $array)
      */
     public static function only($array, $keys)
     {
-        return array_intersect_key($array, array_flip((array) $keys));
+        $attributeData = [];
+
+        foreach ((array) $keys as $key) {
+            $keyValue = data_get($array, $key, '__missing__');
+
+            if ($keyValue == '__missing__') {
+                continue;
+            }
+
+            data_set($attributeData, $key, $keyValue);
```

Comment:
Ah, right. So you could use `Arr::get/set` in this modified `Arr::only`. It would be a much better fit, while still working with the code in Validator.

This modifed `Arr::only` is probably slower, that's the only caveat I can think of.